Expense/expenseRouter.js

```diff
@@ -0,0 +1,32 @@
+const express = require('express');
+const Expense = require('./Expense.js');
+
+const router = express.Router();
+
+router
+    .route('/')
+    .get((req, res) => {
+        Expense.find()
+        .populate('budget', 'title budgetAmount')
+        .then(expenseObj => {
+            res.status(200).json(expenseObj);
+        })
+        .catch(err => {
+            res.status(500).json({ err: 'Expense not found.'});
+        })
+    })
+    .post((req, res) => {
```

Comment:
Make sure we're fetching then saving our budget and category `ObjectId` to our new Expense. Below's an example of how you could go about it with `Promise.all`

Postman `req.body`:
```
{
	"amount": 35,
	"description": "potatoes",
	"budget": "Monthly Spending",
	"category": "Groceries"
}
```

Expense `POST` route:
```
.post((req, res, next) => {
  let { amount, budget, category, description } = req.body

  budget = Budget.findOne({ title: budget })
  category = Category.findOne({ title: category })

  Promise.all([budget, category])
    .then(([budget_id, category_id]) => {
      const expense = new Expense({
        amount,
        budget: budget_id,
        category: category_id,
        description
      })

      expense.save()
        .then(response => res.status(201).json(response))
        .catch(err => next(err))
    })
    .catch(err => next(err))
  })
```

We have to get `budget` and `category` with the `findOne` method, then using `Promise.all`, we grab the `ObjectId` for each from the response from MongoDB, then create our new Expense.